components/camel-jms/src/main/java/org/apache/camel/component/jms/JmsBinding.java

```diff
@@ -467,7 +467,7 @@ protected Message createJmsMessage(Exchange exchange, Object body, Map<String, O
         // special for transferExchange
         if (endpoint != null && endpoint.isTransferExchange()) {
             LOG.trace("Option transferExchange=true so we use JmsMessageType: Object");
-            Serializable holder = DefaultExchangeHolder.marshal(exchange);
+            Serializable holder = DefaultExchangeHolder.marshal(exchange, false, endpoint.isAllowSerializedHeaders());
```

Comment:
Still avoid serializing exchange props, as they may contain some instances of non serializable classes, i.e. for exchange history tracking purposes.
BTW if needed it could be easily exposed an additional parameter to control even this behavior.